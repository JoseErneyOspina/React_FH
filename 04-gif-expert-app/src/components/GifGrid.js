import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';
import useFetchGifs from '../hooks/useFetchGifs';

const GifGrid = ( { category } ) => {

    // Desestructuramos data que viene del custom hook
    const { data:images, loading } = useFetchGifs( category );


    return (
        <>
            <h3 className = 'animate__animated animate__fadeIn'> { category } </h3>
            { loading && <p className = 'animate__animated animate_flash'>Loading</p> }
            <div className ='card-grid'>
                {
                    images.map( img => (
                        <GifGridItem
                            key = { img.id }
                            // img = { img }
                            // Podemos usar el siguiente metodo u usar props en el componente
                            { ...img } // AsÃ­ envio cada una de las propiedades como una propiedad independiente
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export  default GifGrid;
