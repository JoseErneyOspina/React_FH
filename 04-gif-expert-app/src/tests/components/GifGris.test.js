import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid  from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
import GifGridItem from './../../components/GifGridItem';

// Con este mocl fingire cualquier llamada a ese archivo y suponer o controlar la informaici√≥n que eso va a responder
jest.mock('../../hooks/useFetchGifs');

describe('Testing on <GifGrid />', () => {

    const category = 'One Punch'
    // Simulo que ya tiene cierta data
    test('should show correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category = { category } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should show items when the images are loaded with useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];
        // Simulo que ya tiene cierta data
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category = { category } /> );
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });

});