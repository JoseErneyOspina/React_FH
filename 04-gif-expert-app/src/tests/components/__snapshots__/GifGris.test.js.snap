// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GifGrid /> should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate_flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
